src/waveform/renderers/waveformmarkrange.h

```diff
@@ -46,4 +47,6 @@ class WaveformMarkRange {
     friend class WOverview;
 };
 
+typedef std::shared_ptr<WaveformMarkRange> WaveformMarkRangePointer;
```

Comment:
I still don't understand why you need a shared pointer at all?? All instances of WaveformMarkRange are stored in internal vectors and you don't pass them around.

Reply:
Well, the crash was happening in destructor of `WaveformMarkRange` called from `m_markRanges.push_back(WaveformMarkRange());`. I do understand why is destructor executed here, but I don't understand why `delete m_markStartPointControl;` was causing the crash, when `m_markStartPointControl` was initialized with `NULL` in constructor. Could it be that constructor didn't get executed for some reason? :confused:

Reply:
So, I thought it would be simpler if I just changed `m_markRanges` to be a vector of pointers to objects.

Reply:
WaveformMarkRange is non-copyable despite having the default copy-constructor generated by the compiler. The COs pointers are deleted twice!!

* Disallow copy construction/assignment
* Implement a proper move constructor